web/packages/teleport/src/services/bot/consts.ts

```diff
@@ -16,9 +16,58 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { ApiBot, FlatBot } from 'teleport/services/bot/types';
+import {
+  ApiBot,
+  FlatBot,
+  GitHubRepoRule,
+  ProvisionTokenSpecV2GitHub,
+} from 'teleport/services/bot/types';
 
-export function makeListBot(bot: ApiBot): FlatBot {
+/**
+ *
+ * @param spec a ProvisionTokenSpecV2GitHub
+ * @returns the server's teleport/api/types.ProvisionTokenSpecV2GitHub,
+ * which has similar properties but different casing
+ */
+export function toApiGitHubTokenSpec(spec: ProvisionTokenSpecV2GitHub | null) {
+  if (!spec) {
+    return null;
+  }
+  return {
+    allow: spec.allow.map(toApiGitHubRule),
+    enterprise_server_host: spec.enterpriseServerHost,
+  };
+}
+
+/**
+ *
+ * @param param0 a GitHubRepoRule
```

Comment:
According to the [docs](https://jsdoc.app/tags-param#parameters-with-properties), we should do something like this:

```ts
/**
*
* @param {Object} rule ...
* @param {string} rule.sub ...
* ...
*/
```

But since we use Typescript, this feels unnecessary. I'll update it to:

```ts
/**
 *
 * @param {GitHubRepoRule} rule a GitHubRepoRule
 */
```